sklearn/utils/_array_api.py

```diff
@@ -45,7 +46,9 @@ def yield_namespaces(include_numpy_namespaces=True):
         yield array_namespace
 
 
-def yield_namespace_device_dtype_combinations(include_numpy_namespaces=True):
+def yield_namespace_device_dtype_combinations(
+    include_numpy_namespaces=True, include_float16=False
```

Comment:
`_array_api.supported_float_dtypes` returns `float16` if it's in the array namespace. The docstring for that method reads:

> `float16` is not officially part of the Array API spec at the time of writing but scikit-learn estimators and functions can choose to accept it when xp.float16 is defined.
> 
> https://data-apis.org/array-api/latest/API_specification/data_types.html

I opted to ensure StandardScaler would work with `float16` inputs as that is a common use-case for me working with GPUs. I figured it could be an incremental change to other estimators as needed.

Reply:
I implemented it as an opt-in kwarg to allow for require tests/callers to opt-in to testing against `float16`. Would a separate method/fixture be more appropriate, you think?